VerilogForVerificationStyle.md

```diff
@@ -0,0 +1,863 @@
+    By using these it is possible to drastically reduce line length
+    and improve readability.
+2.  `downcast(<extended_class_handle>, <base_class_handle>)`
+    *   Use this macro to cast a handle for any base class object that
+    contains a handle to an extended class to the extended class type, useful
+    for situations that require polymorphism.
+3.  `uvm_object_new` and `uvm_component_new`
+    *   Use these macros to include the `new()` constructor for classes
+    derived from `uvm_object` and `uvm_component`, respectively.
+    If any additional logic must be included in the constructor,
+    these macros must not be used, and a constructor must be written manually.
+4.  `uvm_create_obj` and `uvm_create_comp`
+    *   Use these macros to call `create(<instance_name>)` and
+    `create(<instance_name>, this)`, respectively.
+5.  `DV_CHECK_*` comparison checking macros
+    *   These macros must be used to perform checks on the input arguments
+    and invoke one of the `uvm_info`, `uvm_error`, or `uvm_fatal`
+    reporting macros if the check fails.
+    These are much more concise than hand-writing the comparison checks and
+    the corresponding UVM report macro and provide efficient readability.
+    *   The severity of the invoked report macro can be specified with the
+    <uvm_severity> argument, the default reporting macro is `uvm_error`.
+    *   Append `_FATAL` to the end of these macro names to invoke variants of
+    these macros that will automatically throw fatal errors upon check failure
+    (e.g. `DV_CHECK_EQ_FATAL`, `DV_CHECK_LE_FATAL`, etc..).
+6.  Randomization macros
+    *   `dv_macros.svh` provides three sets of macros which must be used for
+    all randomization functionality:
+        *   `DV_CHECK_RANDOMIZE_FATAL` - Shorthand for `foo.randomize()`
+        with a `uvm_fatal` check
+        *   `DV_CHECK_STD_RANDOMIZE_FATAL` - Shorthand for
+        `std::randomize(foo)` with a `uvm_fatal` check
+        *   `DV_CHECK_MEMBER_RANDOMIZE_FATAL` - Shorthand for
+        `this.randomize(foo)` with a `uvm_fatal` check
+    *   Variants of all three macros that allow constraints to be specified
+    also exist, these macros are invoked by inserting `WITH` into the macro name:
+    `DV_..._RANDOMIZE_WITH_FATAL`
+7.  `DV_EOT_PRINT_..._CONTENTS` TLM fifo display macros
+    *   It is recommended to use these macros to display the status of the
+    various TLM fifos and queues in the scoreboard at the end of the test
+    to ensure that those data structures have been emptied. There should be no
+    pending transactions that have not yet been compared before the simulation
+    ends.
+    If used, these macros must be used during the `check_phase`.
+8.  `DV_SPINWAIT`
+    *   It is recommended to use this macro in situations that require an
+    isolation fork to disable some process after waiting for a specified
+    amount of time, as it provides a built in timer watchdog and proper process isolation:
```

Comment:
recommend "proper process isolation such as the example below."

Reply:
Thanks for the pointer, fixed.